### PR TITLE
chore: add story with buttons to track sequence indicator container width

### PR DESCRIPTION
**Pull request by SethDonohue**
# Description

## Motivation & context

Add story to Carousel to showcase that the sequence indicator container should not be overlapping any actions in the content corners of the carousel. This is partially dependent on the implementation of the carousel content (as the width of the indicators must be reduced), but is described in the story so it is clearly known what to look for when viewing the story.
This story is associated with PR #2405 which fixed the sequence indicator container going full width across the carousel and covering up content.

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-react-msft/assets/carousel-hero-content.tsx [hunk 1]
```diff
@@ -46,16 +45,31 @@ export default class CarouselHeroContent extends React.Component<
...
         },
+        extraButtons: false,
     };
 
     private imageSrc: string =
         this.props.theme === CarouselSlideTheme.light
             ? "http://placehold.it/1399x600/2F2F2F/171717"
             : "http://placehold.it/1399x600/";
 
+    private extraButtonRightStyle: any = {
```
Comment (chrisdholt) [inline packages/fast-components-react-msft/assets/carousel-hero-content.tsx]: Are these actually *any*? (and all the way down)

File: packages/fast-components-react-msft/src/carousel/carousel.stories.tsx [hunk 2]
```diff
@@ -3,163 +3,87 @@ import React from "react";
...
+                    }
+                    theme={theme}
+                    extraButtons={extraButtons}
+                />
+            ),
+        });
+    }
+    return fillerArray;
+}
+
```
Comment (nicholasrice) [inline packages/fast-components-react-msft/src/carousel/carousel.stories.tsx]: Can you ensure the types are accurate here? A number of `any` types.

File: packages/fast-components-react-msft/assets/carousel-hero-content.tsx [hunk 3]
```diff
@@ -1,18 +1,16 @@
-import React from "react";
+import React, { Fragment } from "react";
 import {
+    Button,
```
Comment (chrisdholt) [inline packages/fast-components-react-msft/assets/carousel-hero-content.tsx]: nit: Can we use something other than button here? If it's "accent", then lets just use `<AccentButton />`

File: packages/fast-components-react-msft/assets/carousel-hero-content.tsx [hunk 4]
```diff
@@ -1,18 +1,16 @@
-import React from "react";
+import React, { Fragment } from "react";
```
Comment (chrisdholt) [inline packages/fast-components-react-msft/assets/carousel-hero-content.tsx]: FYI - You're already bringing in React, you *could* reference here as `React.Fragment` without the import.

File: packages/fast-components-react-msft/src/carousel/carousel.stories.tsx [hunk 5]
```diff
@@ -3,163 +3,87 @@ import React from "react";
...
+const carouselSequenceStylesOverrides: any = {
```
Comment (chrisdholt) [inline packages/fast-components-react-msft/src/carousel/carousel.stories.tsx]: Can we type this?

File: packages/fast-components-react-msft/src/carousel/carousel.stories.tsx (hunk 2 above)
Comment (SethDonohue) [reply packages/fast-components-react-msft/src/carousel/carousel.stories.tsx]: Yeah, sorry spaced on that.

File: packages/fast-components-react-msft/assets/carousel-hero-content.tsx [hunk 6]
```diff
@@ -81,13 +101,13 @@ export default class CarouselHeroContent extends React.Component<
                         }}
                     >
                         <Heading
-                            tag={HeadingTag.h2}
-                            size={HeadingSize._3}
+                            tag={this.props.heading.tag}
+                            size={this.props.heading.size}
```
Comment (janechu) [inline packages/fast-components-react-msft/assets/carousel-hero-content.tsx]: heading is optional, this will throw an error if it is not passed.

**Pull request merged**